### PR TITLE
Create course

### PR DESCRIPTION
**Pull request by jennrobles**

### CODE REVIEW

File: client/src/pages/createnewcourse/CreateNewCourse.jsx [hunk 1]
```diff
@@ -1,38 +1,139 @@
...
+        });
+        setLessons(newLessons);
+    };
+
+    const handleCreateCourse = async () => {
+
+        setUploading(true);
+
+        try {
+            const uploadPromises = lessons.map(lesson => videoService.uploadVideo(lesson.selectedFile, lesson.title));
```
Comment (nickerick) [inline client/src/pages/createnewcourse/CreateNewCourse.jsx]: The required workflow is a bit convoluted, it's documented in the PRs that added these but going off the API alone isn't obvious. The `videoUrl` part of a course is only relevant for non self-hosted videos. E.g., a youtube video. If it's self-hosted, then we query that video ourself and the `videoUrl` can be ignored.

You need to create a course, then lessons, THEN upload videos.

The videos are queried in a format of `c{courseId}.l{lessonId}.mp4`. This means when you upload a file to S3, the filename needs to be converted into that format. That's also why you must create a course and lessons first - so that you know what to name the videos you're uploading. Note this [loadVideo](https://github.com/nickerick/skill-magnet/blob/d974789ace74e276d4942449f5d4dcebd3398b93/client/src/pages/courseviewer/CourseViewer.jsx#L29C3-L33C5) method on the course viewer screen. I originally had this in the example screen but I removed it when testing something and apparently forgot to change it back.

So basically after you create all the lessons, you need to map through list of lessons and do something like

```
lessons.map((lesson) =>videoService.uploadVideo(`c{createdCourse.id}.l{lesson.id}.mp4`);
```

File: client/src/pages/createnewcourse/CreateNewCourse.jsx [hunk 2]
```diff
@@ -1,38 +1,139 @@
...
+                category: courseInfo.category
+            };
+
+            const createdCourse = await CourseService.createCourse(coursePayload);
+
+            const lessonPromises = lessons.map((lesson, index) => {
+                const lessonPayload = {
+                    title: lesson.title,
+                    courseId: createdCourse.id,
+                    videoLink: videoUrls[index],
```
Comment (nickerick) [inline client/src/pages/createnewcourse/CreateNewCourse.jsx]: This line + the `videoUrls` can be completely deleted
